fdbserver/VersionedBTree.actor.cpp

```diff
@@ -6433,14 +6675,12 @@ class VersionedBTree {
 
 			// If the children changed, replace [cBegin, cEnd) with newLinks
 			if (u.childrenChanged) {
+				// TODO(yiwu): do we need to clone in all cases?
+				cloneForUpdate();
```

Comment:
I thought we had talked about this function taking a Cursor &c to modify with SwitchTree if it actually made a change?

Reply:
There's one call site inside CommitSubtree which I'm not sure which cursor it is using. Plus if we can remove the `buildNewSubtree` logic (which if it is doable I will address in a separate PR) , the change here is not needed, and in that case I will revert the change here in that PR.

Reply:
Sounds good.